src/consensus/aft/raft.h

```diff
@@ -102,10 +102,14 @@ namespace aft
     bool is_execution_pending = false;
     std::list<std::unique_ptr<AbstractMsgCallback>> pending_executions;
 
-    // When we receive append entries from a new primary, we should rollback
-    // to the index they're telling us about, and replay from here. However
-    // for complex reasons we want to do this once only, and expect to be
-    // correct without doing this thereafter.
+    // When this node receives append entries from a new primary, it may need to
+    // roll back an uncommitted suffix it holds. It must trust the new primary
+    // to dictate this index, and trust it to be honest about its approximation
```

Comment:
I don't think approximation of committed state is the right description here, the new primary sends its committable suffix, and does not make a statement about whether that is committed or not.